src/SAWScript/Crucible/MIR/Setup/Value.hs

```diff
@@ -119,6 +123,22 @@ data MirPointer sym tp = MirPointer
     , _mpRef :: MirReferenceMux sym tp
     }
 
+-- | A slice-related MIR 'SetupValue'.
+data MirSetupSlice
+  = MirSetupSliceRaw (MS.SetupValue MIR) (MS.SetupValue MIR)
+    -- ^ A \"raw\" slice constructed directly from a pointer and a length.
+    -- Currently, this is only used by @crucible-mir-comp@. SAWScript offers no
+    -- way to use this, although we may consider doing so in the future.
+  | MirSetupSlice (MS.SetupValue MIR)
+    -- ^ A slice of a reference to a contiguous sequence 'SetupValue'.
+    -- Currently, this only supports references to arrays.
+  | MirSetupSliceRange (MS.SetupValue MIR) Int Int
+    -- ^ A slice of a reference to a contiguous sequence 'SetupValue', where the
+    -- slice only covers the range specified by the given start and end values
+    -- (the first and second 'Int', respectively). Currently, this only
+    -- supports references to arrays, and it only supports concrete ranges.
```

Comment:
An important new data type that classifies all of the different ways one can construct slice values in SAW/`crucible-mir-comp`.